gpu/containers/src/device_memory.cpp

```diff
@@ -174,11 +174,43 @@ void pcl::gpu::DeviceMemory::upload(const void *host_ptr_arg, std::size_t sizeBy
     cudaSafeCall( cudaDeviceSynchronize() );
 }
 
+bool
+pcl::gpu::DeviceMemory::upload(void* host_ptr_arg,
+                               std::size_t device_begin_byte_offset,
+                               std::size_t num_bytes)
+{
+  const void* const begin = static_cast<char*>(data_) + device_begin_byte_offset;
+  const char* const upload_end = static_cast<const char*>(begin) + num_bytes;
+  const char* const array_end = static_cast<char*>(data_) + sizeBytes_;
+  if (upload_end > array_end) {
+      return false;
+  }
+  cudaSafeCall(cudaMemcpy(host_ptr_arg, begin, num_bytes, cudaMemcpyHostToDevice));
```

Comment:
I think `host_ptr_arg` and `begin` have to be switched (compare other upload function)

Reply:
Hm..., I have to think about that but at first glance I would say you are right... Strange that this didn't result in an error when I thought I tested it. Thanks for pointing this out!